### PR TITLE
executor: implement NULL DEFINED BY for LOAD DATA

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: ref #40499

Problem Summary:

### What is changed and how it works?

as title

This PR also fix that ASCII 00 can set as FIELDS ENCLOSED BY or FIELDS ESCAPED BY

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Comment by lance6716** [comment]
ptal @gozssky @buchuitoudegou

**Review by buchuitoudegou — APPROVED** [review]

File: br/pkg/lightning/mydump/csv_parser.go [hunk 1]
```diff
@@ -208,16 +208,13 @@ func (parser *CSVParser) unescapeString(input field) (unescaped string, isNull b
...
 		unescaped = unescape(unescaped, "", parser.escFlavor, parser.escapedBy[0], parser.unescapeRegexp)
 	}
-	if len(parser.quote) == 0 || !parser.quotedNullIsText {
-		isNull = parser.escFlavor != escapeFlavorMySQLWithNull &&
-			!parser.cfg.NotNull &&
-			slices.Contains(parser.cfg.Null, unescaped)
-	} else if !input.quoted {
+	if !(len(parser.quote) > 0 && parser.quotedNullIsText && input.quoted) {
+		// this branch represents "quote is configured && quotedNullIsText && this field has no quote"
 		// quoted string can never be NULL except for \N, which must be escapeFlavorMySQLWithNull
```
Comment (lance6716) [inline br/pkg/lightning/mydump/csv_parser.go:213]:
```suggestion
		// this branch represents "quote is not configured" or "quoted null is null" or "this field has no quote"
		// we check null for them
```

File: parser/ast/dml.go [hunk 2]
```diff
@@ -1953,32 +1953,32 @@ type FieldItem struct {
 // FieldsClause represents fields references clause in load data statement.
 type FieldsClause struct {
 	Terminated  string
-	Enclosed    byte
-	Escaped     byte
+	Enclosed    *byte
+	Escaped     *byte
```
Comment (gozssky) [inline parser/ast/dml.go:1957]: Why do we need to change the type to a pointer?

**Review by gozssky — APPROVED** [review]

**Pull request merged**